### DISCUSSION TITLE
How to log files on SDCard Deck?

### DISCUSSION THREAD

**Discussion by Rwik2000** [bitcraze/discussions#498] [Q&A]
I tried the following link from the documentation:
https://www.bitcraze.io/products/micro-sd-card-deck/

I am unable to toggle `usd.logging` variable between 1/0. How do we change the variable's value while the crazyflie is on? 

Here is my config.txt file:

~~~
25 # buffer size in bytes
log_ # file name
1 # enable on startup (0/1)
on:fixedFrequency
250 # frequency
1 # mode (0: disabled, 1: synchronous stabilizer, 2: asynchronous)

acc.x
acc.y
acc.z
gyro.x
gyro.y
gyro.z
baro.asl
baro.temp
baro.pressure 
~~~

Even though, I enable the startup variable, but it seems it is not able to log.

**Comment by knmcguire** [comment]
The config.txt parser is very simple so there might be something in there it can not handle.

Have you tried to remove the extra enter between #mode and the first variable

**Comment by Rwik2000** [comment]
I will try to remove the white space.
How do I toggle the usd.logging variable? Do I need to turn it to 0, in order to save the log files? If so, how do I do that?

**Comment by knmcguire** [comment]
Ah sorry I misunderstood your question as a bug report, so I've converted your issue to a discussion.

You toggle the usb.logging variable through the parameter framework, which you can do from the cfclient. You can open up a parameter tab:
- https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#:~:text=used%20input%20interface-,Tabs%20and%20toolboxes,-The%20main%20interface
- https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/parameters_tab/

**Reply by Rwik2000** [reply]
I am still having problems with logging. Whenever, I put the SD card in crazyflie, the CF starts for a few seconds and then I see the front 2 leds flashing red and then the crazyflie restarts. I am not able to log on to the SD Card.

**Reply by krichardsson** [reply]
It looks like the start of your config is wrong, the version is missing for instance. You can check out the [example file](https://github.com/bitcraze/crazyflie-firmware/blob/2022.12/tools/usdlog/config.txt) to see what it should look like. 
I'm not sure but I think the order of fields is important and also that all of them are added.

**Reply by Rwik2000** [reply]
Can you tell me a brief step by step procedure of what all to do.

**Reply by krichardsson** [reply]
Check out the "Usage" tab in the [SD-card deck product page](https://www.bitcraze.io/products/micro-sd-card-deck/)
